models/Asset/MetaData/MetaDataTrait.php

```diff
@@ -0,0 +1,341 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @category   Pimcore
+ * @package    Property
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Model\Asset\MetaData;
+
+use Pimcore\Tool;
+
+trait MetaDataTrait
+{
+    /**
+     * @param bool $force
+     * @return array
+     * @throws \Exception
+     */
+    public function getEmbeddedMetaData(bool $force){
+        if($force){
+            $this->handleEmbeddedMetaData(false);
+        }
+        return $this->getCustomSetting('meta-information') ? : [];
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function handleEmbeddedMetaData(bool $noExif = false) {
+        if(!$this->getCustomSetting('readMetaData') || $this->getDataChanged()){
+            $this->readEmbeddedMetaData($noExif);
+        }
+    }
+    
+    /**
+     * @return array
+     * @throws \Exception
+     */
+    private function readEmbeddedMetaData(bool $noExif = false) : array {
```

Comment:
We should give `$noExif` a more meaningful name, what about `bool $useExifTool = true`